pylint/checkers/classes.py

```diff
@@ -503,6 +504,27 @@ def visit_functiondef(self, node):
         except astroid.NotFoundError:
             pass
 
+        if len(node.body) == 1 and isinstance(node.body[0], (Expr, Return)):
+            value = node.body[0].value
+            super_func = value.parent.parent  # TODO: How do you get access to the method being overriden?
+            func = value
+            # If the args passed to this function is exactly the same, this is a normal override
+            # If they've changed the args, then the behaviour has changed.
+            # if name == "__init__" then it can be a Expr, otherwise we're changing the behaviour by not returning
+            # This only detects if the way super is called is exactly the same as the parent signature
+            # If it isn't - it's likely that changes are being made
+            value_args = [getattr(x, "name", None) for x in func.args]
+            parent_args = [getattr(x, "name", None) for x in super_func.args.args]
+            value_kwargs = [getattr(x, "name", None) for x in func.kwargs]
+            parent_kwargs = [getattr(x, "name", None) for x in super_func.args.kwargs]
+            print value
```

Comment:
Done.

On Sat, 18 Jun 2016 at 00:27 moylop260 <notifications@github.com> wrote:

> In pylint/checkers/classes.py
> <https://github.com/PyCQA/pylint/pull/941#discussion_r67518687>:
>
> > @@ -503,6 +504,27 @@ def visit_functiondef(self, node):
> >          except astroid.NotFoundError:
> >              pass
> >
> > +        if len(node.body) == 1 and isinstance(node.body[0], (Expr, Return)):
> > +            value = node.body[0].value
> > +            super_func = value.parent.parent  # TODO: How do you get access to the method being overriden?
> > +            func = value
> > +            # If the args passed to this function is exactly the same, this is a normal override
> > +            # If they've changed the args, then the behaviour has changed.
> > +            # if name == "__init__" then it can be a Expr, otherwise we're changing the behaviour by not returning
> > +            # This only detects if the way super is called is exactly the same as the parent signature
> > +            # If it isn't - it's likely that changes are being made
> > +            value_args = [getattr(x, "name", None) for x in func.args]
> > +            parent_args = [getattr(x, "name", None) for x in super_func.args.args]
> > +            value_kwargs = [getattr(x, "name", None) for x in func.kwargs]
> > +            parent_kwargs = [getattr(x, "name", None) for x in super_func.args.kwargs]
> > +            print value
>
> Please, remove the print
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/PyCQA/pylint/pull/941/files/c280157eaeb8a79ee51338407dc47e75caecb745#r67518687>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AARehdwLaxmWUOrvOTxzJFIyyMySDWEhks5qMq7igaJpZM4I4XsN>
> .
>